content/articles/building-scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/index.md

```diff
@@ -0,0 +1,115 @@
+
+## Table of contents
+The tutorial will cover:
+- Why is the multi-module approach better than the monolith
+- Advantages of multi-module over microservices architecture
+- Setting a Spring Boot multi-module project
+- How can we make them interact with each other?
+- Gaining the benefits of microservices architecture from the multi-module architecture
+- Bonus
+- Conclusion
+
+## Prerequisites
+To make the most of this tutorial, it is required to have:
+•	Basic understanding of microservices.
+•	Familiarity with the Spring Boot framework.
+•	Good internet connection
+•	Intellij code editor installed.
+
+### Why is the multi-module better than monolith?
+Traditional, monolith design puts all the services for a project in one singular module. Everything is housed in one big box just as they are. In multi-module, each service is compartmentalised before being placed in the big box; the project. Multi-module allows for code re-usability unlike monolith. Code maintenance and adding new features or functionality is much easier.
+
+### Advantages of Multi-module over Microservices
+No doubt, the microservices architecture is beneficial. However, in comparison with the multi-module architecture, it appears to have some short-comings. The first would be ensuring availability of services at optimal time. Because services in the microservices architecture are deployed separately, it means that a downtime in one service could cripple an entire organisation, depending on how many services depend on it. Also, passing around dependencies can be tricky. For large teams and projects, this can be challenging. With the multi-module architecture, services are still developed individually but within a mother project that caters to all of them. Therefore, interest is not necessarily spread across several services but is channelled to the mother project and by extension, the child services. As a result, multi-module projects are easier to maintain than microservices.
+
+### Actual Setup of a Multi-Module project
+Now we go through the process of setting up a Spring Boot multi-module project. We will mimic a small hospital management system and see how different departments will interact in terms of services. To build this, we would need some services, say `Doctor Service`, `Patient Service`, `Consultation Service`. while these can co-exist in separate modules according to their name, we will create modules as recommended the spring way – data, service and controller modules - all together in the same project - `demohospital`. The method for creating these four modules can be used to create as many as are needed in the project.
+![Multi Module in Picture](/engineering-education/building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven/picone.jpg)
```

Comment:
`building-Scalable-projects-using-the-multi-module-architecture-with-spring-boot-and-maven`

Make `S` lowercase for all the image paths

Reply:
Done accordingly for all of the cases invloved